src/lib/utils/options/options.ts

```diff
@@ -72,6 +74,10 @@ export class Options extends ChildableComponent<Application, OptionsComponent> {
         this.trigger(event);
         this.setValues(event.data, '', event.addError.bind(event));
 
+        if (event.compilerOptions) {
+            _.assign(this.compilerOptions, event.compilerOptions);
```

Comment:
Should we use `event.data` for everything instead of having two ways to pass this data?